src/pages/admin/transactions.tsx

```diff
@@ -103,6 +104,8 @@ const Transactions = () => {
   const { data, totalPages, totalTransactions, page } =
     transactionResponse ?? {};
 
+  const [paginationNumber, setPaginationNumber] = useState<number>(page ?? 1);
```

Comment:
The queryParams should serve as the source of truth. i.e Normalize the `pageQuery` to a number
```suggestion
  const page = pageQuery ? parseInt(pageQuery) ?? 1 : 1
```
The `page` value being passed as `pageQuery` to `useGetTransactions` should use the normalized `page` value instead.

The reason for this is that if `page` in `useQuery` is always resolved to the same number for empty pageQuery and first page, clicking page 1 (in the UI) wouldn't cause a refetch after the page has been loaded.

Basically `useGetTransactions` query Keys should remain unchanged for both
`/transactions` and
`/transactions?page=1`
(would have to change `page` type in `useGetTransactions` from `string | null` to `number`)

Another good side effect is you don't need to manage pagination in useState.
As the pagination component sets the page in pageQuery on pagination change, and since currentPage uses the normalized pageQuery `page` it will always be updated.

(note: no need to destructure page from the response)